adam-core/src/test/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDSuite.scala

```diff
@@ -176,6 +176,68 @@ class AlignmentRecordRDDSuite extends ADAMFunSuite {
     }
   }
 
+  sparkTest("round trip with single CRAM file produces equivalent Read values") {
+    val readsPath = resourcePath("artificial.cram")
+    val referencePath = resourceUrl("artificial.fa").toString
+    sc.hadoopConfiguration.set(CRAMInputFormat.REFERENCE_SOURCE_PATH_PROPERTY,
+      referencePath)
+
+    val ardd = sc.loadBam(readsPath)
+    val rddA = ardd.rdd
+
+    val tempFile = tmpFile("artificial.cram")
+    ardd.saveAsSam(tempFile,
+      asType = Some(SAMFormat.CRAM),
+      asSingleFile = true,
+      isSorted = true)
+
+    val rddB = sc.loadBam(tempFile)
+
+    assert(rddB.rdd.count() === rddA.rdd.count())
+
+    val readsA = rddA.rdd.collect()
+    val readsB = rddB.rdd.collect()
+
+    readsA.indices.foreach {
```

Comment:
Ah, I typically do a collect before the zip, which eliminates said issue (and we need to collect to use asserts anyways).